### PR TITLE
s3_bucket: add integration tests

### PR DESCRIPTION
**Pull request by julienvey**
Also update testing-policies/storage

##### SUMMARY

Some improvements to the s3_bucket module (such as #35110) will require an upgrade to boto3. Before trying to do the upgrade, this PR add integration tests for the current version of the module

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

s3_bucket

##### ANSIBLE VERSION

```
2.6.0
```

### CODE REVIEW

**Comment by julienvey** [comment]
@s-hertel @mattclay this PR will requires ci update with the updated permissions described in https://github.com/julienvey/ansible/blob/87fe580fafb14b3d7af7041fbd4146ec2935f3ea/hacking/aws_config/testing_policies/storage-policy.json

**Comment by julienvey** [comment]
bot_status

**Comment by s-hertel** [comment]
Hey, @julienvey, appreciate all the work you've been doing on AWS modules. These can be in their own target since if a change is made to aws_s3 it doesn't need to run the s3_bucket integration tests and vice versa. I had a pull request open for these that has languished out of general disinterest and these look nicer so I'm going to close that. But FYI for the future, https://ansible.sivel.net/pr/byfile.html is a really excellent tool so you can see if you're duplicating effort somewhere.

**Comment by julienvey** [comment]
@s-hertel thank you for the link, much simpler than querying github :)

I will move the tests into their own target

**Comment by s-hertel** [comment]
Shippable is failing right now due to lack of CI permissions. I'll look at updating those later today or next week if I can't get to it.

File: test/integration/targets/s3_bucket/tasks/main.yml
Comment (s-hertel) [inline test/integration/targets/s3_bucket/tasks/main.yml:18]: Anywhere that's using resource_prefix in the bucket name needs to be changed to {{ resource_prefix | lower }} because resource_prefix is not necessarily lowercase (such as, on a mac it may be ansible-test-shertel-OSX-12345678 which fails). Would also be good to have a test that a bucket's policy can be modified.

Comment (s-hertel) [reply test/integration/targets/s3_bucket/tasks/main.yml:18]: I have a CI pull request open for each of your pull requests needing CI update, including this one. Those will hopefully be reviewed/merged soon.

Comment (willthames) [reply test/integration/targets/s3_bucket/tasks/main.yml:18]: My view is that resource_prefix should be fixed at source rather than here.

Comment (julienvey) [reply test/integration/targets/s3_bucket/tasks/main.yml:18]: Done. 

Thank you for the CI updates.

**Comment by willthames** [comment]
@julienvey @s-hertel I've raised https://github.com/ansible/ansible/pull/37095 to fix the `resource_prefix` case - I'd prefer to merge that than adding `|lower` to all uses of resource_prefix.

File: test/integration/targets/s3_bucket/tasks/main.yml
Comment (julienvey) [reply test/integration/targets/s3_bucket/tasks/main.yml:18]: @willthames I only saw your comment after I pushed my changes

**Comment by julienvey** [comment]
@willthames Looks good to me. I can revert my latest changes to remove the `lower` filter, and just wait for your PR to be merged

**Comment by julienvey** [comment]
@willthames Removed the `lower` filter. Good to go once #37095 is merged

File: test/integration/targets/s3_bucket/tasks/main.yml
Comment (s-hertel) [reply test/integration/targets/s3_bucket/tasks/main.yml:18]: Thanks @willthames

**Review by s-hertel — APPROVED** [review]

**Review by ryansb — APPROVED** [review]

**Pull request merged**
